Add az-formdata rule to suggest check on formData parameters (#36)
Azure/azure-api-style-guide: docs/azure-ruleset.md

All `4xx` and `5xx` responses should specify `x-ms-error-response: true` except for `404` response of HEAD operation.

### az-formdata

Check for appropriate use of formData parameters.

It can be appropriate to use formData parameters when sending multiple file-type parameters or an array of file-type parameters.

> But it is unnecessary and overly complicated to use formData to send a single file-type parameter.

Reviewer: ```suggestion
But it is usually unnecessary and can be overly complicated to use formData when all you are doing is sending a single file-type parameter. 
```

> This should be defined as a `body` parameter with `type: string, format: binary` and use content-type `application/octet-stream`.

Reviewer: ```suggestion
Instead, consider defining a `body` parameter with `type: string, format: binary` and use content-type `application/octet-stream`.
```
